### PR TITLE
Convert release app to using components

### PR DESCRIPTION
**Pull request by vanitabarrett**
Trello: https://trello.com/c/G2qXHdHq/96-convert-release-app-to-using-components

This is quite a big change - I recommend reviewing commit by commit to break it up into smaller chunks.

# Homepage Before and After
Screen Shot 2019-11-29 at 16 36 53
<hr>
Screen Shot 2019-11-29 at 16 33 42

# Application Page Before and After
Screen Shot 2019-11-29 at 16 37 07
<hr>
Screen Shot 2019-11-29 at 16 33 59

# Example form in Release app, Before and After
Screen Shot 2019-11-29 at 16 37 22
<hr>
Screen Shot 2019-11-29 at 16 34 07

# Deployment List Before and After
Screen Shot 2019-11-29 at 16 37 41
<hr>
Screen Shot 2019-11-29 at 16 34 24

### CODE REVIEW

File: app/views/applications/edit.html.erb [hunk 1]
```diff
@@ -10,11 +10,12 @@
       }
     ]
   } %>
-  <h1 class="remove-bottom-margin">
```
Comment (andysellick) [inline app/views/applications/edit.html.erb]: Was there any CSS associated with this class?

File: app/views/applications/index.html.erb [hunk 2]
```diff
@@ -6,8 +6,14 @@
...
 } %>
 
-<h1>Applications</h1>
+<%= render "govuk_publishing_components/components/title", {
+  title: "Applications"
+} %>
 
 <%= render partial: "applications_table", locals: { applications: @applications, environments: @environments } %>
 
-<%= link_to 'Create application', new_application_path, class: "btn btn-success" %>
```
Comment (andysellick) [inline app/views/applications/index.html.erb]: At some point we should think about whether this button should be all the way at the bottom of the page, but that's probably out of scope for this PR.

File: app/views/applications/new.html.erb [hunk 3]
```diff
@@ -8,7 +8,13 @@
     }
   ]
 } %>
-<h1>Create new application</h1>
-<p class="lead">Applications are created automatically when first deployed, but you can create them manually if you like.</p>
```
Comment (andysellick) [inline app/views/applications/new.html.erb]: Is there any CSS associated with this class that needs removing?

File: app/views/applications/show.html.erb [hunk 4]
```diff
@@ -32,12 +33,20 @@
...
-  <p>This application has been marked as archived.</p>
+  <%= render "govuk_publishing_components/components/notice", {
+    title: "This application has been marked as archived.",
+    margin_bottom: 4,
+  } %>
 <% end %>
 
 <%= render 'status_notes', application: @application %>
 
-<h2 class="add-bottom-margin">Commit log</h2>
```
Comment (andysellick) [inline app/views/applications/show.html.erb]: Is there any CSS that needs deleting here?

File: app/views/applications/show.html.erb [hunk 5]
```diff
@@ -89,17 +98,26 @@
     </tbody>
   </table>
 <% else %>
-  <div class="alert alert-error">
-    <div>Couldn't get data from GitHub:</div>
-    <div><%= @github_error %></div>
+  <%= render "govuk_publishing_components/components/govspeak", {
+  } do %>
+    <div class="application-notice help-notice">
```
Comment (andysellick) [inline app/views/applications/show.html.erb]: Not sure where this appears, but is this the most suitable component for this?

File: app/views/applications/stats.html.erb [hunk 6]
```diff
@@ -10,11 +10,12 @@
       }
     ]
   } %>
-  <h1 class="remove-bottom-margin">
-    <span class="name"><%= @application.name %> stats</span>
-    <span class="shortname">(<%= @application.shortname %>)</span>
-    <%= render partial: "provider_badge", locals: { application: @application } %>
-  </h1>
+  <%= render "govuk_publishing_components/components/title", {
+    title: "#{@application.name} Stats",
```
Comment (andysellick) [inline app/views/applications/stats.html.erb]: `Stats` here is now capitalised, is that intentional?

File: app/views/stats/index.html.erb [hunk 7]
```diff
@@ -7,10 +7,15 @@
     }
   ]
 } %>
-<h2>Deployments per month</h2>
+
+<%= render "govuk_publishing_components/components/title", {
+  title: "Deployments per month"
+} %>
```
Comment (andysellick) [inline app/views/stats/index.html.erb]: This is probably a difference between live and integration, but I noticed that the page title appears slightly wrong on the stats page. It should be `Deployment stats | GOV.UK Release` but it's coming out as `Deployment stats | GOV.UK GOV.UK Release` (also I can't see what's changed that could have caused this).

Screenshot 2019-12-02 at 10 35 59

File: app/views/stats/index.html.erb [hunk 8]
```diff
@@ -7,10 +7,15 @@
...
 
 <%= line_chart @stats.per_month, width: "100%", height: "450px", legend: false, download: true %>
 
-<h2>Deployments per year</h2>
+<%= render "govuk_publishing_components/components/title", {
```
Comment (andysellick) [inline app/views/stats/index.html.erb]: Spacing here is a bit tight...

Screenshot 2019-12-02 at 10 40 54

File: app/views/applications/edit.html.erb (hunk 1 above)
Comment (vanitabarrett) [reply app/views/applications/edit.html.erb]: Nope, it perhaps came from `govuk_admin_template`

https://github.com/alphagov/govuk_admin_template/search?q=remove-bottom-margin&unscoped_q=remove-bottom-margin

File: app/views/applications/index.html.erb (hunk 2 above)
Comment (vanitabarrett) [reply app/views/applications/index.html.erb]: Agreed, that makes sense to consider as a separate piece of work.

File: app/views/applications/new.html.erb (hunk 3 above)
Comment (vanitabarrett) [reply app/views/applications/new.html.erb]: Nope, possibly a bootstrap thing? There are some later commits which remove bootstrap and any unused CSS.

File: app/views/applications/show.html.erb (hunk 4 above)
Comment (vanitabarrett) [reply app/views/applications/show.html.erb]: No, again it looks like a `govuk_admin_template` thing https://github.com/alphagov/govuk_admin_template/search?q=add-bottom-margin&unscoped_q=add-bottom-margin

File: app/views/applications/show.html.erb (hunk 5 above)
Comment (vanitabarrett) [reply app/views/applications/show.html.erb]: This is what it looks like:

Screenshot 2019-12-02 at 11 42 29

Happy for suggestions for alternatives

**Comment by vanitabarrett** [comment]
@andysellick Thanks, looking at your comments now 👍 In terms of the tabs component, the [commit](https://github.com/alphagov/release/pull/501/commits/6f5a6a2e2170c4f3c8f5959ba818a0e94945cd04) explains why I haven't used the tabs components. Essentially, they're designed to be used to switch between content on the page, whereas the tabs in the Release app are misleading as they actually just go to a new URL. Changing that functionality seemed like quite a big piece of work.

File: app/views/stats/index.html.erb (hunk 7 above)
Comment (vanitabarrett) [reply app/views/stats/index.html.erb]: Fixed. Turns out the admin layout component uses the `browser_title` and `product_name` that you pass it to [generate the title](https://github.com/alphagov/govuk_publishing_components/blob/master/app/views/govuk_publishing_components/components/_layout_for_admin.html.erb#L7). I'd set the product name to "GOV.UK Release" which was leading to a duplicate "GOV.UK".

**Comment by vanitabarrett** [comment]
@alex-ju I've updated the tabs following our conversation on Slack, if you'd like to take a look

**Comment by andysellick** [comment]
Well, I only reviewed the first two commits, so hopefully the rest was okay... 🤞

**Comment by alex-ju** [comment]
@andysellick: oh, I thought you went through it. Please do carry on if you have time and I or Vanita can raise a separate PR for the requested changes.

**Comment by ChrisBAshton** [comment]
This is looking nice! Just spotted the changes today.

**Pull request merged**